x-pack/plugins/security_solution/public/resolver/store/data/selectors.test.ts

```diff
@@ -175,19 +182,28 @@ describe('data state', () => {
         // receive a different databaseDocumentID. this should cause the middleware to abort the existing request and start a new one
         {
           type: 'appReceivedNewExternalProperties',
-          payload: { databaseDocumentID: secondDatabaseDocumentID },
+          payload: {
+            databaseDocumentID: secondDatabaseDocumentID,
+            resolverComponentInstanceID: resolverComponentInstanceID2,
+          },
         },
       ];
     });
     it('should be loading', () => {
       expect(selectors.isLoading(state())).toBe(true);
     });
     it('should need to fetch the second databaseDocumentID', () => {
-      expect(selectors.databaseDocumentIDToFetch(state())).toBe(secondDatabaseDocumentID);
+      expect(selectors.databaseDocumentIDToFetch(state())).toBe(firstDatabaseDocumentID);
     });
     it('should need to abort the request for the databaseDocumentID', () => {
       expect(selectors.databaseDocumentIDToFetch(state())).toBe(secondDatabaseDocumentID);
     });
+    it('should use the correct location for the first resolver', () => {
+      expect(selectors.resolverComponentInstanceID(state())).toBe(resolverComponentInstanceID1);
```

Comment:
❔ How does this work? It looks like we expect different things back to back from the same selector...

Reply:
❔  If it's because of the aborted request, that's really hard to understand here.

Reply:
It fetched the first resolver, and then this is it resolved. Technically the second one is`in-progress` at this point and resolves in the checks below